tools/stacksnoop.lua

```diff
@@ -100,12 +100,8 @@ return function(BPF, utils)
 
   local TASK_COMM_LEN = 16 -- linux/sched.h
 
-  bpf:get_table("events"):open_perf_buffer(print_event, [[
-    struct {
-      uint64_t stack_id;
-      uint32_t pid;
-      char comm[%d];
-    }
-  ]] % {TASK_COMM_LEN})
+  bpf:get_table("events"):open_perf_buffer(print_event,
+    "struct { uint64_t stack_id; uint32_t pid; char comm[$]; }",
+    TASK_COMM_LEN)
```

Comment:
Right. And it is also typesafe. You can read the link to the FFI Semantics page I posted early -- it explains parameterized types in detail. :)